src/arch/x86/process.cc

```diff
@@ -580,6 +580,18 @@ X86_64Process::initState()
             tc->setMiscReg(MISCREG_CR0, cr0);
 
             tc->setMiscReg(MISCREG_MXCSR, 0x1f80);
+
+            // Setting CR3 to the pid so that concatinated page addr with
+            // lower 12 bits of CR3 can be used in SE mode as well to
+            // avoid conflicts between tlb entries with same virtual
+            // addresses belonging to different processes
+            tc->setMiscReg(MISCREG_CR3, pTable->pid());
+
+            // Setting pcide bit in CR4
+            CR4 cr4 = 0;
+            cr4.pcide = 1;
+            tc->setMiscReg(MISCREG_CR4, cr4);
```

Comment:
Is this not going to overwrite line 379? I think we might as well put `cr4.pcide = 1;` up there

Reply:
It should not overwrite because line 379 is part of the block which executes when KVM CPU is used (KVMinSE is true). I guess, we should set both CR3 (with pid) and CR4.pcide for KVM CPU as well. Do you think that will break anything for KVM CPU?